src/pages/homeCollection/kitCSV.js

```diff
@@ -0,0 +1,133 @@
+import { nonUserNavBar } from "../../navbar.js";
+import { homeCollectionNavbar } from "./homeCollectionNavbar.js";
+import { showAnimation, hideAnimation, getIdToken, baseAPI, keyToNameObj, keyToNameAbbreviationObj, triggerErrorModal, convertISODateTime } from "../../shared.js";
+import { activeHomeCollectionNavbar } from "./activeHomeCollectionNavbar.js";
+import { conceptIds } from '../../fieldToConceptIdMapping.js';
+import { receiptedCSVFileTemplate, downloadCSVfile } from '../receipts/csvFileReceipt.js';
+
+export const kitCsvScreen = (auth) => {
+  const user = auth.currentUser;
+  if (!user) return;
+  const username = user.displayName ? user.displayName : user.email;
+  showAnimation();
+  kitCsvTemplate(username);
+  csvFileButtonSubmit();
+  hideAnimation();
+};
+
+const kitCsvTemplate = (name) => {
+  let template = ``;
+  template += homeCollectionNavbar();
+
+  template += `<div id="root root-margin" style="margin-top:3rem;">
+              <div id="alert_placeholder"></div>
+              ${receiptedCSVFileTemplate()}
+              </div>`
+  
+  document.getElementById("contentBody").innerHTML = template;
+  document.getElementById("navbarNavAltMarkup").innerHTML = nonUserNavBar(name);
+  activeHomeCollectionNavbar();
+};
+
+const csvFileButtonSubmit = () => {
+  document.getElementById("csvCreateFileButton").addEventListener("click", async (e)=> {
+      e.preventDefault();
+      const dateFilter = document.getElementById("csvDateInput").value + 'T00:00:00.000Z'; 
```

Comment:
Better call it 'dateString' or something like this.

Reply:
You can see the name caused confusion ( Object.keys() was used on its value passed to API).